clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -795,12 +795,29 @@ static void addSanitizers(const Triple &TargetTriple,
     PB.registerOptimizerLastEPCallback(SanitizersCallback);
   }
 
-  if (LowerAllowCheckPass::IsRequested()) {
+  bool lowerAllowCheck = LowerAllowCheckPass::IsRequested();
+  // Is there a non-zero cutoff?
+  static constexpr double SanitizerMaskCutoffsEps = 0.000000001f;
+  for (unsigned int i = 0; i < SanitizerKind::SO_Count; ++i) {
+    lowerAllowCheck |= (CodeGenOpts.SanitizeSkipHotCutoffs[i].value_or(0) >
+                        SanitizerMaskCutoffsEps);
+  }
+
+  if (lowerAllowCheck) {
```

Comment:
Instead of having this loop, maybe we can intruduce:
`std::optional<vector<int>> SanitizeSkipHotCutoffs::getAllScaled(int Num) const {}`

Reply:
Done